_use-release-management.shade

```diff
@@ -0,0 +1,191 @@
+                continue;
+            }
+
+            try
+            {
+                GitCommand(repo, "log -1 --exit-code origin/dev..origin/release");
+            }
+            catch
+            {
+                Log.Warn("Unmerged changes in repository " + repo);
+                GitCommand(repo, "log origin/dev..origin/release");
+                throw;
+            }
+        }
+
+        Log.Info("No conflicts in repos, continuing with creating release branch.");
+
+        foreach (var repo in GetAllRepos())
+        {
+            GitCommand(repo, "checkout origin/dev -B release");
+
+            // Update NuGet.Config
+            var nugetConfigPath = Path.Combine(repo, "NuGet.config");
+            if (File.Exists(nugetConfigPath))
+            {
+                var original = File.ReadAllText(nugetConfigPath);
+                var modified = original
+                    .Replace("https://www.myget.org/F/aspnetcidev", "https://www.myget.org/F/aspnetcirelease")
+                    .Replace("https://www.myget.org/F/azureadwebstacknightly", "https://www.myget.org/F/azureadwebstackrelease");
+
+                if (!string.Equals(original, modified, StringComparison.Ordinal))
+                {
+                    File.WriteAllText(nugetConfigPath, modified);
+                    GitCommand(repo, "add NuGet.config");
+                    GitCommand(repo, "commit -m \"Updating to release NuGet.config.\"");
+                }
+            }
+
+            GitCommand(repo, "push origin release:release");
+            GitCommand(repo, "checkout origin/dev -B dev");
+            GitCommand(repo, "merge release -s ours");
+            GitCommand(repo, "push origin dev:dev");
+        }
+    }
+    
+#update-master .pull-all
+    -// Merge release branch to master
+    -// Pin versions of packages in project.json and updated project.lock.json
+    -// More information https://github.com/aspnet/Universe/wiki/%23pin-version-:-Pinning-package-version-for-a-particular-release-in-project.json
+    @{
+
+        if (string.IsNullOrEmpty(coherenceFeed))
+        {
+            throw new Exception("COHERENCE_FEED not specified. Usually this is Packages-NoTimestamp directory of Coherence-Signed.");
+        }
+
+        if (string.IsNullOrEmpty(koreBuildVersion))
+        {
+            throw new Exception("KOREBUILD_VERSION environment variable is not specified.");
+        }
+
+        if (string.IsNullOrEmpty(nugetVersion))
+        {
+            throw new Exception("NUGET_VERSION not specified. This is most recent stable build of NuGet from http://dist.nuget.org/index.html. e.g. v3.2");
+        }
+
+        var excludeReposForJson = new[]
+        {
+          "Coherence",
+          "Coherence-Signed",
+          "dnvm",
+          "Entropy",
+          "Setup",
+          "libuv-build",
+        };
+
+        Exec("cmd", "/C dnu restore", @"tools\PinVersion");
```

Comment:
Yeah I just copied the code around so `makefile.shade` is easier to navigate. We're going to a dry run of release next week and I'll clean up the tools at that point.